lib/src/client/http2_connection.dart

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) 2018, the gRPC project authors. Please see the AUTHORS file
+// for details. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:http2/transport.dart';
+import 'package:meta/meta.dart';
+
+import '../shared/timeout.dart';
+
+import 'call.dart';
+import 'connection.dart' hide ClientConnection;
+import 'connection.dart' as connection;
+
+import 'transport/http2_credentials.dart';
+import 'transport/http2_transport.dart';
+import 'transport/transport.dart';
+
+class Http2ClientConnection implements connection.ClientConnection {
+  static final _methodPost = new Header.ascii(':method', 'POST');
+  static final _schemeHttp = new Header.ascii(':scheme', 'http');
+  static final _schemeHttps = new Header.ascii(':scheme', 'https');
+  static final _contentTypeGrpc =
+      new Header.ascii('content-type', 'application/grpc');
+  static final _teTrailers = new Header.ascii('te', 'trailers');
+  static final _grpcAcceptEncoding =
+      new Header.ascii('grpc-accept-encoding', 'identity');
+  static final _userAgent = new Header.ascii('user-agent', 'dart-grpc/0.2.0');
+
+  final ChannelOptions options;
+
+  connection.ConnectionState _state = ConnectionState.idle;
+
+  @visibleForTesting
```

Comment:
Is this necessary, if it's no longer part of the public interface?

The `Http2ClientConnection` class is only available if you import from `package:grpc/src/...` right?
(not that it hurts, I guess it's nice to document that this is only here for testing purposes :)

Reply:
Good points. You are right that I made it visibleForTesting before I decided to remove the ClientConnection from the public interface. But it is strictly there for testing, so I guess I'll keep it.